src/amo/components/RatingManager/index.js

```diff
@@ -182,40 +190,91 @@ export class RatingManagerBase extends React.Component<InternalProps, State> {
     }
   };
 
-  render() {
-    const { i18n, addon, userId, userReview, version } = this.props;
-    const { showTextEntry } = this.state;
-    const isLoggedIn = Boolean(userId);
+  cancelTextEntry = (event: SyntheticEvent<any>) => {
+    event.preventDefault();
+    this.setState({ showTextEntry: false });
+  };
 
-    invariant(addon, 'addon is required');
-    invariant(version, 'version is required');
+  renderTextEntry() {
+    const { _config, userReview } = this.props;
+
+    if (!userReview || !this.isSignedIn()) {
+      return null;
+    }
+
+    if (_config.get('enableInlineAddonReview')) {
+      return <AddonReviewManager review={userReview} />;
+    }
+
+    return (
+      <AddonReview
+        onReviewSubmitted={this.onReviewSubmitted}
+        review={userReview}
+      />
+    );
+  }
+
+  isSignedIn() {
+    return Boolean(this.props.userId);
+  }
+
+  renderUserRatingForm() {
+    const { i18n, addon, userReview } = this.props;
 
     const prompt = i18n.sprintf(
       i18n.gettext('How are you enjoying your experience with %(addonName)s?'),
       { addonName: addon.name },
     );
 
+    return (
+      <form action="">
+        <fieldset>
+          <legend className="RatingManager-legend">{prompt}</legend>
+          <div className="RatingManager-ratingControl">
+            {!this.isSignedIn() ? this.renderLogInToRate() : null}
+            <UserRating
+              readOnly={!this.isSignedIn()}
+              onSelectRating={this.onSelectRating}
+              review={!this.isSignedIn() ? null : userReview}
+            />
+          </div>
+        </fieldset>
+      </form>
+    );
+  }
+
+  renderInlineReviewControls() {
+    const { i18n } = this.props;
+    const { showTextEntry } = this.state;
+
+    return showTextEntry && this.isSignedIn() ? (
```

Comment:
Should there be a test for the case when `showTextEntry` is true but the user is not signed in, or do we consider that to be not possible?

Reply:
Actually, it is possible. You can begin editing your review and then--without saving it--click the *log out* button at the top of the page. This state is covered here: https://github.com/mozilla/addons-frontend/pull/5997/files#diff-5d954e9f650d2beb87d2ca0344afe003R502